lib/multi-acc-new-eks-mixed-observability-pattern/pipeline.ts

```diff
@@ -0,0 +1,388 @@
+
+    let ArgoCDAddOnProps: blueprints.ArgoCDAddOnProps;
+
+    if (repoType.toLocaleLowerCase() === 'public') {
+        ArgoCDAddOnProps = {
+            bootstrapRepo: {
+                repoUrl: repoUrl,
+                path: path,
+                targetRevision: branch,
+            },
+        };
+    } else {
+
+        ArgoCDAddOnProps = {
+            bootstrapRepo: {
+                repoUrl: repoUrl,
+                path: path,
+                targetRevision: branch,
+                credentialsSecretName: 'github-ssh-key', // for access to private repo. This needs SecretStoreAddOn added to your cluster. Ensure github-ssh-key secret exists in pipeline account at COA_REGION
+                credentialsType: 'SSH',
+            },                     
+        };      
+    }    
+    return new blueprints.ArgoCDAddOn(ArgoCDAddOnProps);
+}
+
+
+function createGOArgoAddonConfig(repoUrl: string, path: string, branch?: string, repoType?: repoTypeValues): blueprints.ArgoCDAddOn {
+
+    branch = branch! || 'main';
+    repoType = repoType! || 'public';
+
+    const ampAssumeRoleArn = `arn:aws:iam::${ampAccount}:role/${ampAssumeRoleName}`;
+    const cwAssumeRoleArn = `arn:aws:iam::${cwAccount}:role/${cwAssumeRoleName}`;
+
+    const ampEndpointURL = "UPDATE_ME_WITH_AMP_ENDPOINT_URL";
+
+    let ArgoCDAddOnProps: blueprints.ArgoCDAddOnProps;
+
+    if (repoType.toLocaleLowerCase() === 'public') {
+        ArgoCDAddOnProps = {
+            bootstrapRepo: {
+                repoUrl: repoUrl,
+                path: path,
+                targetRevision: branch,
+            },
+        };
+    } else {
+
+        ArgoCDAddOnProps = {
+            bootstrapRepo: {
+                repoUrl: repoUrl,
+                path: path,
+                targetRevision: branch,
+                credentialsSecretName: 'github-ssh-key', // for access to private repo. This needs SecretStoreAddOn added to your cluster. Ensure github-ssh-key secret exists in pipeline account at COA_REGION
+                credentialsType: 'SSH',
+            },                     
+        };       
+    }
+
+    ArgoCDAddOnProps.bootstrapValues = {
+        AMP_ASSUME_ROLE_ARN: ampAssumeRoleArn,
+        AMP_AWS_REGION: ampRegion,
+        CW_ASSUME_ROLE_ARN: cwAssumeRoleArn,
+        CW_AWS_REGION: cwRegion,        
+        AMP_ENDPOINT_URL: ampEndpointURL,
+        AMG_ENDPOINT_URL: amgWorkspaceUrl,
+        GRAFANA_CLUSTER_DASH_URL: clusterDashUrl,
+        GRAFANA_KUBELET_DASH_URL: kubeletDashUrl,
+        GRAFANA_NSWRKLDS_DASH_URL: namespaceWorkloadsDashUrl,
+        GRAFANA_NODEEXP_DASH_URL: nodeExporterDashUrl,
+        GRAFANA_NODES_DASH_URL: nodesDashUrl,
+        GRAFANA_WORKLOADS_DASH_URL: workloadsDashUrl
+    };
+
+    // ArgoCDAddOnProps.values = {
+    //     server: {  // By default argocd-server is not publicaly exposed. uncomment this section, if you need to expose using ALB
+    //         service: {
+    //             type: 'LoadBalancer'
```

Comment:
remove comments

Reply:
I've kept this comment for future reference in case if argocd service need to be front-ended with ALB. Please let me know if you still prefer to remove this comment.